src/redisearch_rs/encode_decode/src/varint/vector_writer.rs

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2006-Present, Redis Ltd.
+ * All rights reserved.
+ *
+ * Licensed under your choice of the Redis Source Available License 2.0
+ * (RSALv2); or (b) the Server Side Public License v1 (SSPLv1); or (c) the
+ * GNU Affero General Public License v3 (AGPLv3).
+*/
+
+use super::write;
+
+#[derive(Debug)]
+pub struct VectorWriter {
+    buffer: Vec<u8>,
+    n_members: usize,
+    last_value: u32,
+}
+
+impl VectorWriter {
+    /// Create a new `VectorWriter` with the given capacity.
+    pub fn new(cap: usize) -> Self {
+        Self {
+            buffer: Vec::with_capacity(cap),
+            last_value: 0,
+            n_members: 0,
+        }
+    }
+
+    /// Write an integer to the vector.
+    ///
+    /// # Return Value
+    ///
+    /// The number of bytes written to the vector.
+    pub fn write(&mut self, value: u32) -> std::io::Result<usize> {
+        self.buffer.reserve_exact(16);
```

Comment:
Using `reserve_exact` because we want to be memory efficient?
I know this is the most similar implementation to what we have now, but personally, I don't trust that a lot of thinking or benchmarking went into the C implementation.

Reply:
I added this to be close to the existing impl, yes. I also doubt it's needed. I'll see if if can be safely removed.